build error

OS: Manjaro 20.2 Nibia
Kernel: x86_64 Linux 5.4.78-1-MANJARO
cmake: 3.18.4
libiconv: 1.16-1
---
```bash
Scanning dependencies of target csv2xls
[ 81%] Building CXX object CMakeFiles/csv2xls.dir/csv2xls.cpp.o
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h: In function ‘TargetType ConvertTo(SourceType)’:
/tmp/csv2xls/csv2xlslib/conversion.h:54:43: error: ‘numeric_limits’ is not a member of ‘std’
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                           ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:54:71: error: ‘::min’ has not been declared; did you mean ‘std::min’?
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                                                       ^~~
      |                                                                       std::min
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:278:5: note: ‘std::min’ declared here
  278 |     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:54:37: error: expected primary-expression before ‘(’ token
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                     ^
/tmp/csv2xls/csv2xlslib/conversion.h:54:43: error: ‘numeric_limits’ is not a member of ‘std’
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                           ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:54:68: error: expected primary-expression before ‘>’ token
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                                                    ^
/tmp/csv2xls/csv2xlslib/conversion.h:54:71: error: ‘::min’ has not been declared; did you mean ‘std::min’?
   54 |              if (source < SourceType(std::numeric_limits<TargetType>::min()))
      |                                                                       ^~~
      |                                                                       std::min
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:278:5: note: ‘std::min’ declared here
  278 |     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:60:43: error: ‘numeric_limits’ is not a member of ‘std’
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                           ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:60:71: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                       ^~~
      |                                                                       std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:60:37: error: expected primary-expression before ‘(’ token
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                     ^
/tmp/csv2xls/csv2xlslib/conversion.h:60:43: error: ‘numeric_limits’ is not a member of ‘std’
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                           ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:60:68: error: expected primary-expression before ‘>’ token
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                    ^
/tmp/csv2xls/csv2xlslib/conversion.h:60:71: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   60 |              if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                       ^~~
      |                                                                       std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:80:41: error: ‘numeric_limits’ is not a member of ‘std’
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                         ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:80:69: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                                                     ^~~
      |                                                                     std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:80:35: error: expected primary-expression before ‘(’ token
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                   ^
/tmp/csv2xls/csv2xlslib/conversion.h:80:41: error: ‘numeric_limits’ is not a member of ‘std’
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                         ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:80:66: error: expected primary-expression before ‘>’ token
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                                                  ^
/tmp/csv2xls/csv2xlslib/conversion.h:80:69: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   80 |             if(source > SourceType(std::numeric_limits<TargetType>::max()) )
      |                                                                     ^~~
      |                                                                     std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:94:42: error: ‘numeric_limits’ is not a member of ‘std’
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                          ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:94:70: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                      ^~~
      |                                                                      std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
In file included from /tmp/csv2xls/csv2xls.cpp:27:
/tmp/csv2xls/csv2xlslib/conversion.h:94:36: error: expected primary-expression before ‘(’ token
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                    ^
/tmp/csv2xls/csv2xlslib/conversion.h:94:42: error: ‘numeric_limits’ is not a member of ‘std’
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                          ^~~~~~~~~~~~~~
/tmp/csv2xls/csv2xlslib/conversion.h:94:67: error: expected primary-expression before ‘>’ token
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                   ^
/tmp/csv2xls/csv2xlslib/conversion.h:94:70: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   94 |             if (source > SourceType(std::numeric_limits<TargetType>::max()))
      |                                                                      ^~~
      |                                                                      std::max
In file included from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/ios:40,
                 from /usr/include/c++/10.2.0/ostream:38,
                 from /usr/include/c++/10.2.0/iostream:39,
                 from /tmp/csv2xls/csv2xls.cpp:26:
/usr/include/c++/10.2.0/bits/stl_algobase.h:300:5: note: ‘std::max’ declared here
  300 |     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
      |     ^~~
make[2]: *** [CMakeFiles/csv2xls.dir/build.make:82: CMakeFiles/csv2xls.dir/csv2xls.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:986: CMakeFiles/csv2xls.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

--- a/csv2xlslib/conversion.h
+++ b/csv2xlslib/conversion.h
@@ -44,9 +44,9 @@ TargetType ConvertTo(SourceType source)
     {
         return source;
     }
-    else if constexpr (!std::is_convertible<SourceType, TargetType>::value)
+    else
     {
-        static_assert("types are not convertible");
+        static_assert(std::is_convertible<SourceType, TargetType>::value, "types are not convertible");
     }
     if constexpr (std::is_signed<TargetType>::value ==
                   std::is_signed<SourceType>::value)

--- a/csv2xlslib/conversion.h
+++ b/csv2xlslib/conversion.h
@@ -29,6 +29,7 @@
 #include <cstddef>
 #include <ios>
 #include <sstream>
+#include <limits>
 
 class BadNumericConversion : public std::out_of_range
 {

